spatial/r2/vector.go

```diff
@@ -54,6 +54,19 @@ func Norm2(p Vec) float64 {
 	return p.X*p.X + p.Y*p.Y
 }
 
+// Unit returns the unit vector colinear to p.
+func Unit(p Vec) Vec {
+	if p.X == 0 && p.Y == 0 {
+		return Vec{}
```

Comment:
Just thinking. This should probably be `Vec{math.NaN(), math.NaN()}`, no? Same in r3.

Reply:
I piggy-backed on other libraries in my field (ROOT, namely) to get the convention I am using here (_ie:_ `Unit(0) == 0`).
happy to go w/ the `NaN`s, though.